### PR TITLE
Aruha 1173

### PR DESCRIPTION
**Pull request by antban**
# Do not reformat incoming messages while publishing

> Zalando ticket : ARUHA-1173

## Description
Right now nakadi is parsing json while validating events and than stores events from parsed format to a kafka format. This may lead to problems user behavior (for example 0.0 and 0 should be treated in the same manner, but it is actually not)
This ticket prevents nakadi from this kind of modifications, removing only meaningless values from the events.

### CODE REVIEW

File: src/test/java/org/zalando/nakadi/domain/BatchItemTest.java [hunk 1]
```diff
@@ -1,14 +1,93 @@
...
+    public void testBatchItemReplacementsWithMetadata() {
+        final JSONObject event = new JSONObject();
+        final JSONObject sourceMetadata = new JSONObject();
+        sourceMetadata.put("eid", UUID.randomUUID().toString());
+        sourceMetadata.put("occurred_at", (new DateTime(DateTimeZone.UTC)).toString());
+        sourceMetadata.put("partition", "0");
+
+        event.put("metadata", sourceMetadata);
+        event.put("foo", "Test data data data");
+        final BatchItem bi = BatchFactory.from("[" + event.toString(2) + "]").get(0);
```
Comment (adyach) [inline src/test/java/org/zalando/nakadi/domain/BatchItemTest.java]: new JSONArray().put(event).toString(2)

File: src/main/java/org/zalando/nakadi/domain/BatchFactory.java [hunk 2]
```diff
@@ -4,78 +4,163 @@
...
+            }
         }
+        return null;
+    }
 
-        if (sb.length() != 0) {
-            batch.add(new BatchItem(sb.toString()));
+    public static List<BatchItem> from(final String events) {
+        final List<BatchItem> batch = new ArrayList<>();
+        int objectStart = locateOpenSquareBracket(events) + 1;
```
[the remark this answers is not in the document]
Comment (antban) [reply src/main/java/org/zalando/nakadi/domain/BatchFactory.java]: @adyach not to parse too much, also, the name is perfectly correct. object start may be only openSquareBracket + 1 and further.

**Comment by antban** [comment]
:+1:

**Comment by antban** [comment]
deploy validation please

**Comment by adyach** [comment]
👍

**Comment by antban** [comment]
deploy validation please

**Comment by antban** [comment]
deploy validation please

**Comment by antban** [comment]
deploy validation please

**Comment by adyach** [comment]
deploy validation please

**Comment by adyach** [comment]
👍

**Comment by antban** [comment]
:+1:

**Comment by antban** [comment]
deploy validation please

**Comment by lmontrieux** [comment]
👍

**Pull request merged**